### PR TITLE
[ADM-906][frontend][feat]: happy paths scenarios for charting feature

### PR DESCRIPTION
**Pull request by mrcuriosity-tw**
## Summary

...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

File: frontend/e2e/pages/metrics/config-step.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { config as configStepData } from '../../fixtures/create-new/config-step'
 import { METRICS_STEP_SAVING_FILENAME } from '../../fixtures';
 import { downloadFileAndCheck } from '../../utils/download';
 import { expect, Locator, Page } from '@playwright/test';
+import { format } from 'e2e/utils/date-time';
```
Comment (guzhongren) [inline frontend/e2e/pages/metrics/config-step.ts:5]: `e2e` is defined in the tsconfig?
if not, PLZ use relative path

File: frontend/e2e/pages/metrics/config-step.ts [hunk 2]
```diff
@@ -252,6 +255,22 @@ export class ConfigStep {
...
+        startDate: format(ranges[i].startDate),
+        endDate: format(ranges[i].endDate),
+      };
+      if (i) {
+        await this.addNewTimeRange();
+        await expect(this.timeRangeRows).toHaveCount(i + 1);
+      }
+      await this.typeInDateRange({ ...dateRange, number: i });
+      await expect(this.fromDateInput.nth(i)).toHaveValue(dateRange.startDate);
+      await expect(this.toDateInput.nth(i)).toHaveValue(dateRange.endDate);
```
Comment (guzhongren) [inline frontend/e2e/pages/metrics/config-step.ts:270]: Follow the POM of e2e, not add assertion in the action method.

File: frontend/e2e/pages/metrics/metrics-step.ts [hunk 3]
```diff
@@ -637,6 +647,25 @@ export class MetricsStep {
...
+    for (const option of (await options.all()).slice(1)) {
+      const isOptionSelected = (await option.getAttribute('aria-selected')) === 'true';
+      if (!isOptionSelected) {
+        await allOption.click();
+        break;
+      }
+    }
+
+    for (const option of (await options.all()).slice(1)) {
+      expect((await option.getAttribute('aria-selected')) === 'true').toBeTruthy();
```
Comment (guzhongren) [inline frontend/e2e/pages/metrics/metrics-step.ts:663]: No assertion in actions

File: frontend/e2e/pages/metrics/metrics-step.ts [hunk 4]
```diff
@@ -779,4 +808,40 @@ export class MetricsStep {
...
+    };
+    for (let i = 0; i < rangeCount; i++) {
+      const currentRange = this.dateRangeViewerOptions.getByLabel(`date range viewer - option ${i}`);
+      if (currentRange.getByTestId('PriorityHighIcon')) {
+        apiIndicators.red++;
+      }
+      // todo green++
+      // if (currentRange.getByTestId('PriorityHighIcon')) {
+      //   apiIndicators.red++
+      // }
```
Comment (guzhongren) [inline frontend/e2e/pages/metrics/metrics-step.ts:827]: ???

File: frontend/e2e/pages/metrics/metrics-step.ts [hunk 5]
```diff
@@ -779,4 +808,40 @@ export class MetricsStep {
...
+    }
+    expect(apiIndicators.red).toBeGreaterThan(0);
+    // expect(apiIndicators.green).toBeGreaterThan(0)
```
Comment (guzhongren) [inline frontend/e2e/pages/metrics/metrics-step.ts:830]: ???

File: frontend/e2e/pages/metrics/metrics-step.ts [hunk 6]
```diff
@@ -779,4 +808,40 @@ export class MetricsStep {
...
+    );
+  }
+
+  async checkSomeApiFailed(rangeCount: number) {
+    await this.checkApiFailedAlertVisible();
+    await this.checkPartialApiFailedTimeRangeIndicator(rangeCount);
+    // await expect(this.page.getByRole('alert').first()).toContainText('Help Information');
+    // await expect(this.page.getByRole('alert').first()).toContainText(
+    //   'The file will expire in 30 minutes, please download it in time.',
+    // );
```
Comment (guzhongren) [inline frontend/e2e/pages/metrics/metrics-step.ts:845]: ???

File: frontend/e2e/pages/metrics/report-step.ts [hunk 7]
```diff
@@ -339,27 +406,98 @@ export class ReportStep {
...
+    csvCompareLines,
+  }: {
+    trigger: Locator;
+    rangeCount: number;
+    csvCompareLines?: ICsvComparedLines;
+  }) {
+    const isNeedToCompareCsvLines = csvCompareLines !== undefined;
+    if (isNeedToCompareCsvLines && rangeCount !== csvCompareLines.length) {
+      throw new Error('given time ranges count and csvCompareLines count are not equal, please check your fixture');
+    }
```
Comment (guzhongren) [inline frontend/e2e/pages/metrics/report-step.ts:457]: Why `throw new Error` in the e2e test case?
there is no throw/if in the the e2e test sceniro

File: frontend/e2e/pages/metrics/report-step.ts [hunk 8]
```diff
@@ -456,4 +609,40 @@ export class ReportStep {
...
+    await this.displayListTab.click();
+    await this.displayChartTab.click();
+
+    expect(await this.displayBoardChartTab.getAttribute('aria-selected')).toEqual('false');
+    expect(await this.displayDoraChartTab.getAttribute('aria-selected')).toEqual('true');
+
+    await expect(this.leadTimeForChangeChart).toBeVisible();
+    await expect(this.deploymentFrequencyChart).toBeVisible();
+    await expect(this.changeFailureRateChart).toBeVisible();
+    await expect(this.meanTimeToRecoveryChart).toBeVisible();
```
Comment (guzhongren) [inline frontend/e2e/pages/metrics/report-step.ts:646]: split it into diff parts.
1. actions like `await this.displayListTab.click();` should not in `check.....`
2. assertion part should be in a function named like `check/validate.....`

Comment (guzhongren) [reply frontend/e2e/pages/metrics/report-step.ts:646]: and the flow of operation should be ordered by the e2e flow, not in this page(report-step)

**Review by guzhongren — CHANGES_REQUESTED** [review]

File: frontend/e2e/pages/metrics/metrics-step.ts (hunk 6 above)
Comment (mrcuriosity-tw) [reply frontend/e2e/pages/metrics/metrics-step.ts:845]: waiting for another PR to merge into, don't worry.

File: frontend/e2e/pages/metrics/metrics-step.ts (hunk 5 above)
Comment (mrcuriosity-tw) [reply frontend/e2e/pages/metrics/metrics-step.ts:830]: Ditto.

File: frontend/e2e/pages/metrics/report-step.ts (hunk 7 above)
Comment (mrcuriosity-tw) [reply frontend/e2e/pages/metrics/report-step.ts:457]: To ensure the range count is equal to `compareCsvLines` if `compareCsvLines` presents.

There IS already some `throw Error` in the e2e test(`e2e/pages/metrics/report-step.ts`)

File: frontend/e2e/pages/metrics/metrics-step.ts (hunk 4 above)
Comment (mrcuriosity-tw) [reply frontend/e2e/pages/metrics/metrics-step.ts:827]: Ditto

File: frontend/e2e/pages/metrics/metrics-step.ts (hunk 3 above)
Comment (mrcuriosity-tw) [reply frontend/e2e/pages/metrics/metrics-step.ts:663]: Didn't get your point

File: frontend/e2e/pages/metrics/config-step.ts (hunk 1 above)
Comment (mrcuriosity-tw) [reply frontend/e2e/pages/metrics/config-step.ts:5]: ok

File: frontend/e2e/pages/metrics/config-step.ts (hunk 2 above)
Comment (zhou-yinyuan) [reply frontend/e2e/pages/metrics/config-step.ts:270]: done!

File: frontend/e2e/pages/metrics/metrics-step.ts (hunk 3 above)
Comment (zhou-yinyuan) [reply frontend/e2e/pages/metrics/metrics-step.ts:663]: done

File: frontend/e2e/pages/metrics/metrics-step.ts (hunk 6 above)
Comment (zhou-yinyuan) [reply frontend/e2e/pages/metrics/metrics-step.ts:845]: done

File: frontend/e2e/pages/metrics/metrics-step.ts (hunk 4 above)
Comment (zhou-yinyuan) [reply frontend/e2e/pages/metrics/metrics-step.ts:827]: done

File: frontend/e2e/pages/metrics/report-step.ts (hunk 7 above)
Comment (guzhongren) [reply frontend/e2e/pages/metrics/report-step.ts:457]: have you fix it???

**Review by guzhongren — CHANGES_REQUESTED** [review]

**Review by guzhongren — APPROVED** [review]

**Review by Mandy-Tang — APPROVED** [review]

**Pull request merged**